E0178 needs to be updated to new format

From: src/test/compile-fail/E0178.rs

E0178 needs a span_label, updating it from:

```
error[E0178]: expected a path on the left-hand side of `+`, not `&'a Foo`
  --> src/test/compile-fail/E0178.rs:14:8
   |
14 |     w: &'a Foo + Copy, //~ ERROR E0178
   |        ^^^^^^^
   |
help: try adding parentheses (per RFC 438):
   |     w: &'a (Foo + Copy), //~ ERROR E0178
```

To:

```
error[E0178]: expected a path on the left-hand side of `+`, not `&'a Foo`
  --> src/test/compile-fail/E0178.rs:14:8
   |
14 |     w: &'a Foo + Copy, //~ ERROR E0178
   |        ^^^^^^^ expected a path
   |
help: try adding parentheses (per RFC 438):
   |     w: &'a (Foo + Copy), //~ ERROR E0178
```

--- a/src/librustc_typeck/astconv.rs
+++ b/src/librustc_typeck/astconv.rs
@@ -1086,6 +1086,7 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
                                                "expected a path on the left-hand side \
                                                 of `+`, not `{}`",
                                                pprust::ty_to_string(ty));
+                err.span_label(ty.span, &format!("expected a path"));
                 let hi = bounds.iter().map(|x| match *x {
                     hir::TraitTyParamBound(ref tr, _) => tr.span.hi,
                     hir::RegionTyParamBound(ref r) => r.span.hi,

--- a/src/test/compile-fail/E0178.rs
+++ b/src/test/compile-fail/E0178.rs
@@ -11,10 +11,18 @@
 trait Foo {}
 
 struct Bar<'a> {
-    w: &'a Foo + Copy, //~ ERROR E0178
-    x: &'a Foo + 'a, //~ ERROR E0178
-    y: &'a mut Foo + 'a, //~ ERROR E0178
-    z: fn() -> Foo + 'a, //~ ERROR E0178
+    w: &'a Foo + Copy,
+    //~^ ERROR E0178
+    //~| NOTE expected a path
+    x: &'a Foo + 'a,
+    //~^ ERROR E0178
+    //~| NOTE expected a path
+    y: &'a mut Foo + 'a,
+    //~^ ERROR E0178
+    //~| NOTE expected a path
+    z: fn() -> Foo + 'a,
+    //~^ ERROR E0178
+    //~| NOTE expected a path
 }
 
 fn main() {

